### ISSUE TITLE
Rename mboxd dbus object to be in correct xyz.openbmc_project

### ISSUE THREAD

**Issue by rfrandse** [ibm-openbmc/dev#118]
<a href="https://github.com/geissonator"><img src="https://avatars2.githubusercontent.com/u/1612261?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [geissonator](https://github.com/geissonator)**
_Wednesday May 24, 2017 at 15:16 GMT_
_Originally opened as https://github.com/openbmc/openbmc/issues/1663_

----

It's currently in org.openbmc.mboxd which is incorrect.  Will require service rename and minor code updates.

FYI - @anoo1 @dkodihal

**Comment by rfrandse** [comment]
<a href="https://github.com/geissonator"><img src="https://avatars2.githubusercontent.com/u/1612261?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [geissonator](https://github.com/geissonator)**
_Friday Jul 14, 2017 at 18:19 GMT_

----

Start with https://gerrit.openbmc-project.xyz/#/c/4739/

**Comment by rfrandse** [comment]
<a href="https://github.com/rfrandse"><img src="https://avatars3.githubusercontent.com/u/15894767?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [rfrandse](https://github.com/rfrandse)**
_Sunday Jul 16, 2017 at 14:46 GMT_

----

abandoned
~~<https://gerrit.openbmc-project.xyz/5318> mboxd: Rename mboxd dbus object to xyz.openbmc_project~~
~~Resolves  #1663 Rename mboxd dbus object to be in correct xyz.openbmc_project~~

**Comment by rfrandse** [comment]
<a href="https://github.com/lgon"><img src="https://avatars1.githubusercontent.com/u/25489748?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [lgon](https://github.com/lgon)**
_Friday Jul 21, 2017 at 14:19 GMT_

----

GitHub Comment
++++++++++++++++++++++++++++++++++++++
TASK LIST
- [x] In phosphor-m boxd repo, add xyz/openbmc_project/mboxd/Internal/Manager.interface.yaml
- [x] In phosphor-mboxd repo, modify configure.ac and Makefile.am in order to use sdbusplus in the build process.
- [x] Implement c++ interfaces defined in .yaml file.
- [x] Modify mboxctl to use new methods that use the sdbusplus bindings.

# **STATUS** - In Review
7-20 - No change in status. Will continue work on this item once #1550 is finished.
7-21 - Started development work, but not much progress yet. Spent most of the day on #1550.
7-24 - No change, was not able to spend time on this today.
7-25 - No change. Was not able to work on this.
7-26 - Started the process outlined in https://w3-connections.ibm.com/wikis/home?lang=en-us#!/wiki/W5d74d2bb8660_4d6e_9bb6_08e4fc6ff375/page/Hello%20World to generate bindings. Spent 1 hour. 
7-27 - Created c++ interface implementatino of .yaml file in the phosphor-mboxd repo, but has compile errors that need to be fixed. The commit is located at https://gerrit.openbmc-project.xyz/#/c/5589/   Spent 5 hours.
7-28 - Continued work  in phosphor-mboxd repo. Modfied Makefile.am, wrote mbox_manager.hpp/cpp, and xyz/openbmc_project/Mboxd/server.hpp files, still working on build issues. Spend 4 hours on this task today.
7-28 - Continued some work, but focused on helping others on the team to get the Delete implementation integrated and tested.
7-29 - Still working on using the mboxd generated code working with the phosphor-mboxd repo.
7-30 - Updated the first 2 commits for the first 2 tasks. https://gerrit.openbmc-project.xyz/#/c/5589/ https://gerrit.openbmc-project.xyz/#/c/5675/
8-1 - Updated Makefile.ac to generate server.hpp and server.cpp based on the .yaml file. Created mboxd_manager.hpp and mboxd_manager.cpp to implement Manager class that subclasses the class generated from the .yaml file.    Still haven't finished implementing all the methods in the .yaml file.
8-2 - Added mboxd::Manager class which inherits from the ServerObject.  Converted mboxd.c from c to c++ and modified it to use an mboxd::Manager object for dbus communication.    Work to finish: Fix the warnings that are occuring after converting c to c++, and finish implementing the details of the methods in mboxd::Manager. Need to fix the failure of the mboxd unit test.
8-3 - Made changes according to code review. Work to do: finish changing mboxd_bus.c to c++ and use the new manager class.
8-10 - Added two new commits. Created a phosphor::mboxd::Dbus class that extends the mboxd sdbusplus bindings, and implemented the ping() method listed in the .yaml. file. Todo: create c++ version of mboxctl to use the new class and method, then finish impementing the rest of the methods.
8-30 - Finished implementation of methods for sdbusplus bindings. Work to do: Finish changes to mboxctl.
9-1 - Pushed out changes for review. Used the example of pnor-partition-table of how to call c++ code from c code in mboxd.
9-5 - Pushed out changes based on comments in the code review. (Fixed issue with file descriptors, removed variables that would only be used in later commits). Work TODO: finish implmenting methods from .yaml file.
9-8 - Finished making changes to mboxctl to use new methods that use the sdbusplus bindings. In Review.
++++++++++++++++++++++++++++++++++++++

**Comment by rfrandse** [comment]
<a href="https://github.com/rfrandse"><img src="https://avatars3.githubusercontent.com/u/15894767?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [rfrandse](https://github.com/rfrandse)**
_Saturday Sep 09, 2017 at 17:57 GMT_

----

Abandoned
~<https://gerrit.openbmc-project.xyz/6578> mboxctl: Implemented mboxctl suspend method~
~Resolves  #1663 Rename mboxd dbus object to be in correct xyz.openbmc_project~

**Comment by rfrandse** [comment]
<a href="https://github.com/geissonator"><img src="https://avatars2.githubusercontent.com/u/1612261?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [geissonator](https://github.com/geissonator)**
_Wednesday Nov 29, 2017 at 16:02 GMT_

----

As long as we filter org out of REST, this is low priority.

**Comment by rfrandse** [comment]
<a href="https://github.com/anoo1"><img src="https://avatars0.githubusercontent.com/u/9052527?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [anoo1](https://github.com/anoo1)**
_Wednesday Nov 29, 2017 at 16:16 GMT_

----

Per Brad, we'll just rename the interface to org.openpower.

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by geissonator** [comment]
Not worth the effort at this point. Lets close this one out.
